examples/chef/chef.py

```diff
@@ -164,40 +236,190 @@ def main(argv: Sequence[str]) -> None:
+        if sys.platform == "darwin":
+            flush_print("Installation of ZAP OS packages not supported on MacOS")
+        if sys.platform == "win32":
+            flush_print(
+                "Installation of ZAP OS packages not supported on Windows")
+
+        flush_print("Running NPM to install ZAP Node.JS dependencies")
+        shell.run_cmd(
+            f"cd {_REPO_BASE_PATH}/third_party/zap/repo/ && npm install")
+
+    #
+    # Populate zzz_generated
+    #
+
+    if options.generate_zzz:
+        flush_print(f"Cleaning {_CHEF_ZZZ_ROOT}")
+        if not os.path.exists(_CHEF_ZZZ_ROOT):
+            flush_print(f"{_CHEF_ZZZ_ROOT} doesn't exist; creating")
+            os.mkdir(_CHEF_ZZZ_ROOT)
+        else:
+            flush_print(f"Deleting and recreating existing {_CHEF_ZZZ_ROOT}")
+            shutil.rmtree(_CHEF_ZZZ_ROOT)
+            os.mkdir(_CHEF_ZZZ_ROOT)
+        flush_print(f"Generating files in {_CHEF_ZZZ_ROOT} for all devices")
+        for device_name in _DEVICE_LIST:
+            flush_print(f"Generating files for {device_name}")
+            device_out_dir = os.path.join(_CHEF_ZZZ_ROOT, device_name)
+            os.mkdir(device_out_dir)
+            device_out_dir = os.path.join(device_out_dir, "zap-generated")
+            os.mkdir(device_out_dir)
+            shell.run_cmd(textwrap.dedent(f"""\
+            {_REPO_BASE_PATH}/scripts/tools/zap/generate.py \
+            {_CHEF_SCRIPT_PATH}/devices/{device_name}.zap -o {device_out_dir}"""))
+            shell.run_cmd(f"touch {device_out_dir}/af-gen-event.h")
+        generate_device_manifest(include_zap_submod=True,
+                                 write_manifest_file=True)
+        exit(0)
+
+    #
+    # CI
+    #
+
+    if options.ci:
+        for device_name in [d for d in _DEVICE_LIST if d in _CI_ALLOW_LIST]:
+            if options.build_target == "nrfconnect":
+                shell.run_cmd("export GNUARMEMB_TOOLCHAIN_PATH=\"$PW_ARM_CIPD_INSTALL_DIR\"")
+            shell.run_cmd(f"cd {_CHEF_SCRIPT_PATH}")
+            shell.run_cmd(f"./chef.py -cbr --use_zzz -d {device_name} -t {options.build_target}")
+        exit(0)
+
+    #
+    # Build all
+    #
+
+    if options.build_all:
+        # Needs testing after refactor
+        # Needs to call per-platform bundle function
+        flush_print("Build all disabled")
+        exit(1)
+        flush_print("Building all chef examples")
+        archive_prefix = "/workspace/artifacts/"
+        archive_suffix = ".tar.gz"
+        os.makedirs(archive_prefix, exist_ok=True)
+        for device_name in _DEVICE_LIST:
+            for platform, platform_meta in cicd_config.items():
+                directory = platform_meta['build_dir']
+                label = platform_meta['platform_label']
+                output_dir = os.path.join(_CHEF_SCRIPT_PATH, directory)
+                command = f"./chef.py -cbr --use_zzz -d {device_name} -t {platform}"
+                flush_print("-" * 64)
```

Comment:
Add a named argument to `flush_print` to optionally print a border. 
Add a per (device X platform) print statement with border to `options.ci`.